New CP member priority parameter description (#504)
hazelcast/hz-docs: docs/modules/cp-subsystem/pages/configuration.adoc

Therefore, it does not improve the degree of fault tolerance compared to five CP
members.

[[configuring-leadership-priority]]
== Configuring Leadership Priority

Some CP members are better leadership candidates than others. For example, members in your primary data center make better leaders because of the reduced latency between clients and the leader. Whereas, members under high load are not good candidates because they are more likely to suffer from downtime. To ensure the availability of the CP subsystem, you can configure CP members with a priority rating, using the <<cp-member-priority, `cp-member-priority`>> option.

Initially, any CP member with any priority can become a leader. But, the leadership rebalancing task periodically transfers the leadership from CP members with the lowest priorities to those with higher priorities within each CP group. Eventually, all CP group leaders are CP members with high priorities.

> WARNING: The leadership rebalancing task works only with members within the CP group. For example, if you start a cluster that has three CP members with priority 0 (with group size also set to three), and you promote a new CP member with priority 100, the new CP member will not participate in the leadership rebalancing until it will not become a member of a CP group.

Reviewer: ```suggestion
NOTE: Only CP members that are part of a CP group can become leaders. CP members that are not in a CP group do not participate in the leadership rebalancing task. For example, you start a cluster with the CP group size set to three and three CP members with priority 0. Later, you promote a new CP member with priority 100. Even though it has a higher priority, the new CP member never becomes a leader because the CP group is full. The new CP member is eligible for leadership only when a CP member leaves the CP group and the new CP member takes the previous member's place.
```

--
====

|[[cp-member-priority]]`cp-member-priority`

> |The CP member priority, provided as a positive or negative integer. The Raft leader's role eventually will be transferred to members with higher priorities within the CP group.

Reviewer: ```suggestion
|The priority rating as a positive or negative integer. The leader's role is eventually transferred to members with higher priorities within a CP group.
```
